tee-worker/ts-tests/integration-tests/data-providers.test.ts

```diff
@@ -0,0 +1,140 @@
+import { randomBytes, KeyObject } from 'crypto';
+import { step } from 'mocha-steps';
+import { assert } from 'chai';
+import { buildIdentityFromKeypair, decryptWithAes, initIntegrationTestContext, PolkadotSigner } from './common/utils';
+import { randomSubstrateWallet } from './common/helpers';
+import { assertIsInSidechainBlock } from './common/utils/assertion';
+import {
+    getSidechainNonce,
+    getTeeShieldingKey,
+    sendRequestFromTrustedCall,
+    createSignedTrustedCallRequestVc,
+} from './common/di-utils'; // @fixme move to a better place
+import type { IntegrationTestContext } from './common/common-types';
+import { CorePrimitivesIdentity, RequestVCResult } from 'parachain-api';
+import { aesKey } from './common/call';
+import { $ as zx } from 'zx';
+import { subscribeToEventsWithExtHash } from './common/transactions';
+import { KeyringPair } from '@polkadot/keyring/types';
+import { u8aToHex } from '@polkadot/util';
+import { vip3CredentialJson, CredentialDefinition } from './common/credential-json';
+import colors from 'colors';
+describe('Test Vc (direct invocation)', function () {
+    let context: IntegrationTestContext = undefined as any;
+    let teeShieldingKey: KeyObject = undefined as any;
+    const substrateIdentities: CorePrimitivesIdentity[] = [];
+
+    const clientDir = process.env.LITENTRY_CLI_DIR;
+    const reqExtHash = '0x0000000000000000000000000000000000000000000000000000000000000000';
+    const keyringPairs: KeyringPair[] = [];
+    let argvId = '';
+    this.timeout(6000000);
+    before(async () => {
+        context = await initIntegrationTestContext(process.env.WORKER_ENDPOINT!, process.env.NODE_ENDPOINT!);
+        teeShieldingKey = await getTeeShieldingKey(context);
+    });
+
+    // usage example:
+    // `pnpm run test-data-providers:local --id=vip3-membership-card-gold` for single test
+    // `pnpm run test-data-providers:local` for all tests
+    const argv = process.argv.indexOf('--id');
+    argvId = process.argv[argv + 1];
+    const {
+        protocol: workerProtocal,
+        hostname: workerHostname,
+        port: workerPort,
+    } = new URL(process.env.WORKER_ENDPOINT!);
+    const { protocol: nodeProtocal, hostname: nodeHostname, port: nodePort } = new URL(process.env.NODE_ENDPOINT!);
+
+    async function linkIdentityViaCli(id: string) {
```

Comment:
is it possible to utilise this function for other VC scenarios, like NFT holder or BNB domain?